src/solver/optimisation/new_constraint_builder.h

```diff
@@ -0,0 +1,400 @@
+#pragma once
+
+#include <antares/logs/logs.h>
+#include "opt_structure_probleme_a_resoudre.h"
+#include "opt_rename_problem.h"
+#include "opt_fonctions.h"
+#include <memory>
+// TODO remove relative include
+#include "../simulation/sim_structure_probleme_economique.h"
+
+#include <variant>
+#include <utility>
+
+namespace NewVariable
+{
+struct SingleIndex
+{
+    explicit SingleIndex(unsigned index) : index(index)
+    {
+    }
+    unsigned index;
```

Comment:
shouldn't we use a more specific and platform independent type such as `uint32_t` ?

Reply:
deprecated following #1617